nifi-nar-bundles/nifi-jms-bundle/nifi-jms-cf-service/src/main/java/org/apache/nifi/jms/cf/JMSConnectionFactoryProvider.java

```diff
@@ -178,7 +178,7 @@ private void setConnectionFactoryProperties(ConfigurationContext context) {
             } else {
                 if (propertyName.equals(BROKER)) {
                     if (context.getProperty(CONNECTION_FACTORY_IMPL).evaluateAttributeExpressions().getValue().startsWith("org.apache.activemq")) {
-                        this.setProperty("brokerURL", entry.getValue());
+                        this.setProperty("brokerURL", context.getProperty(descriptor).evaluateAttributeExpressions().getValue());
```

Comment:
Thank you for your comments. I'll address the ELSE and account for the possibility of null.